pkg/config/data.go

```diff
@@ -111,29 +126,115 @@ var FilePath = func() string {
 // DefaultEndpoint is the default Fastly API endpoint.
 const DefaultEndpoint = "https://api.fastly.com"
 
-// File represents all of the configuration parameters that can end up in the
-// config file. At some point, it may expand to include e.g. user profiles.
+// File represents our dynamic application toml configuration.
 type File struct {
-	Token            string `toml:"token"`
-	Email            string `toml:"email"`
-	Endpoint         string `toml:"endpoint"`
-	LastVersionCheck string `toml:"last_version_check"`
+	Fastly      ConfigFastly              `toml:"fastly"`
+	CLI         ConfigCLI                 `toml:"cli"`
+	User        ConfigUser                `toml:"user"`
+	Language    ConfigLanguage            `toml:"language"`
+	StarterKits ConfigStarterKitLanguages `toml:"starter-kits"`
+}
+
+// ConfigFastly represents fastly specific configuration.
+type ConfigFastly struct {
+	APIEndpoint string `toml:"api_endpoint"`
+}
+
+// ConfigCLI represents CLI specific configuration.
+type ConfigCLI struct {
+	RemoteConfig string `toml:"remote_config"`
+	TTL          string `toml:"ttl"`
+	LastChecked  string `toml:"last_checked"`
+}
+
+// ConfigUser represents user specific configuration.
+type ConfigUser struct {
+	Token string `toml:"token"`
+	Email string `toml:"email"`
+}
+
+// ConfigLanguage represents C@E language specific configuration.
+type ConfigLanguage struct {
+	Rust ConfigRust `toml:"rust"`
+}
+
+// ConfigLanguage represents Rust C@E language specific configuration.
+type ConfigRust struct {
+	// ToolchainVersion is the `rustup` toolchain string for the compiler that we
+	// support
+	ToolchainVersion string `toml:"toolchain_version"`
+
+	// WasmWasiTarget is the Rust compilation target for Wasi capable Wasm.
+	WasmWasiTarget string `toml:"wasm_wasi_target"`
+
+	// FastlySysConstraint is a free-form semver constraint for the internal Rust
+	// ABI version that should be supported.
+	FastlySysConstraint string `toml:"fastly_sys_constraint"`
+}
+
+// ConfigStarterKitLanguages represents language specific starter kits.
+type ConfigStarterKitLanguages struct {
+	AssemblyScript []ConfigStarterKit `toml:"assemblyscript"`
+	Rust           []ConfigStarterKit `toml:"rust"`
+}
+
+// ConfigStarterKit represents starter kit specific configuration.
+type ConfigStarterKit struct {
+	Name   string `toml:"name"`
+	Path   string `toml:"path"`
+	Tag    string `toml:"tag"`
+	Branch string `toml:"branch"`
+}
+
+// Load gets the configuration file from the CLI API endpoint and encodes it
+// from memory into config.File.
+func (f *File) Load(configEndpoint string, httpClient api.HTTPClient) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
3 seconds seems very high? Also can we move this timeout to a named constant at the top of the file?

Reply:
@phamann considering we expect the response to be a lot quicker, is there any real danger in setting it to three seconds? In the case of a freak scenario where the response is slow I'd rather give the client a better chance at acquiring the file rather than setting it too low.